engine/runtime/src/main/java/org/enso/interpreter/runtime/Module.java

```diff
@@ -198,6 +216,27 @@ public void setLiteralSource(String source) {
     setLiteralSource(Rope.apply(source), null);
   }
 
+  /**
+   * Attaches names of submodules that should be accessible from this module
+   *
+   * @param names fully qualified names of (potentially virtual) modules' names
+   */
+  public void setDirectSubmoduleNames(QualifiedName[] names) {
```

Comment:
How can we make sure this value doesn't change more than once? It shouldn't change, right?

Reply:
It shouldn't, added an assert that will blow it up if someone changes the logic.